### PR TITLE
Adds migrator lambda

### PR DESCRIPTION
**Pull request by kenoir**
### What is this PR trying to achieve?

In order to move existing content into the archive service easily, a mechanism to automate ingestion of bagged content (from the `bagger` service) is desirable.

This lambda acts as a bridge between the `bagger` service and the archive ingest api.

### Who is this change for?

👓 🌾

### CODE REVIEW

File: archive/migrator/src/migrator.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+def lambda_handler(event, _):
+    ingest_api_url = os.getenv("INGEST_API_URL")
+
+    for record in event["Records"]:
+        message = record["Sns"]["Message"]
+        decoded_message = json.loads(message)
+
+        print(decoded_message)
+
+        if decoded_message["upload_location"]:
```
Comment (alexwlchan) [inline archive/migrator/src/migrator.py]: Do you want to log something if there isn't an upload_location? Is that an expected/support use case?

File: archive/migrator/src/migrator.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    for message in archive_bag_api_messages(bags, bucket_name):
+        response = session.post(api, json=message)
+        status_code = response.status_code
+
+        if status_code != 201:
+            print_result(f"ERROR calling {api}", response)
+        else:
+            print(f"{message} -> {api} [{status_code}]")
+            location = response.headers.get("Location")
```
Comment (alexwlchan) [inline archive/migrator/src/migrator.py]: Don't we always expect the Location header to be present? Should we error here if it's absent?

**Comment by kenoir** [comment]
@alexwlchan in both cases I agree, but I want to stop working on this for now though as i've got a few hundred bags archived. We'll need to revisit this for the migration so i don't want to lose the work.

**Comment by kenoir** [comment]
That said, both those comments are useful for a later refactor.

**Pull request merged**